### PR TITLE
feat: add a preferred nickname to users

### PR DESCRIPTION
**Pull request by jimsynz**
Allows a signed in user to set their preferred username in the user settings. Prefilled in the "join" form in the lobby.

### CODE REVIEW

File: lib/playdate/accounts/user.ex [hunk 1]
```diff
@@ -6,196 +6,210 @@ defmodule Playdate.Accounts.User do
   alias Playdate.Accounts.{Changes, Preparations, UserToken, Validations}, warn: false
 
   actions do
-    defaults [:read]
+    defaults([:read])
```
Comment (mbuhot) [inline lib/playdate/accounts/user.ex:9]: There’s a lot of parens added in this file. Did something change with the auto formatter? I’ve had to disable the vs-code formatter as it occasionally refuses to use the plugins from mix deps.

File: lib/playdate_web/live/user_settings_live.ex [hunk 2]
```diff
@@ -187,6 +212,28 @@ defmodule PlaydateWeb.UserSettingsLive do
     end
   end
 
+  def handle_event("change_nickname", %{"user" => params}, socket) do
+    socket.assigns.current_user
+    |> AshForm.for_update(:change_nickname, api: Accounts, as: "user")
+    |> AshForm.validate(params)
+    |> AshForm.submit()
```
Comment (mbuhot) [inline lib/playdate_web/live/user_settings_live.ex:219]: I like how submit let’s the UI dispatch an action here 👌

**Review by mbuhot — APPROVED** [review]
👍 if possible revert the parens churn.

File: lib/playdate/accounts/user.ex (hunk 1 above)
Comment (jimsynz) [reply lib/playdate/accounts/user.ex]: Good catch. Thanks!

**Pull request merged**